osu.Game/Screens/Play/HUD/MatchScoreDisplay.cs

```diff
@@ -0,0 +1,164 @@
+                new Box
+                {
+                    Name = "top bar red (static)",
+                    RelativeSizeAxes = Axes.X,
+                    Height = bar_height / 4,
+                    Width = 0.5f,
+                    Colour = colours.TeamColourRed,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopRight
+                },
+                new Box
+                {
+                    Name = "top bar blue (static)",
+                    RelativeSizeAxes = Axes.X,
+                    Height = bar_height / 4,
+                    Width = 0.5f,
+                    Colour = colours.TeamColourBlue,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopLeft
+                },
+                score1Bar = new Box
+                {
+                    Name = "top bar red",
+                    RelativeSizeAxes = Axes.X,
+                    Height = bar_height,
+                    Width = 0,
+                    Colour = colours.TeamColourRed,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopRight
+                },
+                score2Bar = new Box
+                {
+                    Name = "top bar blue",
+                    RelativeSizeAxes = Axes.X,
+                    Height = bar_height,
+                    Width = 0,
+                    Colour = colours.TeamColourBlue,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopLeft
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.X,
+                    Height = font_size + bar_height,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopCentre,
+                    Children = new Drawable[]
+                    {
+                        Score1Text = new MatchScoreCounter
+                        {
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre
+                        },
+                        Score2Text = new MatchScoreCounter
+                        {
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre
+                        },
+                    }
+                },
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Team1Score.BindValueChanged(_ => updateScores());
+            Team2Score.BindValueChanged(_ => updateScores());
+        }
+
+        private void updateScores()
+        {
+            Score1Text.Current.Value = Team1Score.Value;
+            Score2Text.Current.Value = Team2Score.Value;
+
+            var winningText = Team1Score.Value > Team2Score.Value ? Score1Text : Score2Text;
+            var losingText = Team1Score.Value <= Team2Score.Value ? Score1Text : Score2Text;
```

Comment:
This technically favours team 1 in case of a tie, however implausible that may be. Not sure if that's actually desired behaviour?

Reply:
Hmm, this was just copied verbatim from the tournament client, but you make a good point.